x-pack/legacy/plugins/fleet/server/libs/policy.ts

```diff
@@ -4,12 +4,72 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { PoliciesRepository, FullPolicyFile } from '../repositories/policies/types';
+import { flatten, uniq } from 'lodash';
+import { Datasource, Policy } from '../../../ingest/server/libs/adapters/policy/adapter_types';
+import { FrameworkUser } from '../adapters/framework/adapter_types';
+import { AgentPolicy, PoliciesRepository } from '../repositories/policies/types';
 
 export class PolicyLib {
   constructor(private readonly policyAdapter: PoliciesRepository) {}
 
-  public async getFullPolicy(policyId: string): Promise<FullPolicyFile | null> {
-    return await this.policyAdapter.getFullPolicy(policyId);
+  private storedDatasourceToAgentStreams(datasources: Datasource[] = []): AgentPolicy['streams'] {
+    return flatten(
+      datasources.map((ds: Datasource) => {
+        return ds.streams.map(stream => ({
+          ...stream.config,
+          id: stream.id,
+          type: stream.input.type as any,
+          output: { use_output: stream.output_id },
+          ...(stream.config || {}),
+        }));
+      })
+    );
+  }
+
+  private outputIDsFromDatasources(datasources: Datasource[] = []): string[] {
+    return uniq(
+      flatten(
+        datasources.map((ds: Datasource) => {
+          return ds.streams.map(stream => stream.output_id);
+        })
+      )
+    ) as string[];
+  }
+
+  public async getFullPolicy(user: FrameworkUser, id: string): Promise<AgentPolicy | null> {
+    const policy = await this.policyAdapter.get(user, id);
+    if (!policy) {
+      return null;
+    }
+    const agentPolicy = {
+      outputs: {
+        ...(
+          await this.policyAdapter.getPolicyOutputByIDs(
+            user,
+            this.outputIDsFromDatasources(policy.datasources)
+          )
+        ).reduce((outputs, { config, ...output }) => {
+          outputs[output.id] = {
+            ...output,
+            type: output.type as any,
+            ...config,
+          };
+          return outputs;
+        }, {} as AgentPolicy['outputs']),
+      },
+      streams: this.storedDatasourceToAgentStreams(policy.datasources),
+    };
+
+    return agentPolicy;
+  }
+
+  public policyUpdated(
```

Comment:
```suggestion
  public async policyUpdated(
```

Reply:
This will need to be async when fully implemented so I would rather not change this